vaapi/vaapicontext.cpp

```diff
@@ -25,10 +25,63 @@
 #include "vaapi/vaapicontext.h"
 
 #include "common/log.h"
+#include "common/common_def.h"
 #include "vaapi/vaapidisplay.h"
 #include "vaapi/vaapiutils.h"
+#include <algorithm>
+#include <vector>
 
 namespace YamiMediaCodec{
+using std::vector;
+static const VAProfile h264ProfileList[] = {VAProfileH264ConstrainedBaseline, VAProfileH264Main, VAProfileH264High};
+
+//Driver may declare support higher profile but don't support lower profile.
+//In this case, higher profile should be created
+static bool checkH264Profile(VAProfile& profile, vector<VAProfile> profileList)
```

Comment:
better use "const vector<VAProfile>& profileList" else it will make a copy of profileList